pkg/irqs/info.go

```diff
@@ -0,0 +1,100 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2020 Red Hat, Inc.
+ */
+
+package irqs
+
+import (
+	"fmt"
+	"io/ioutil"
+	"path/filepath"
+	"sort"
+	"strconv"
+	"strings"
+
+	"k8s.io/kubernetes/pkg/kubelet/cm/cpuset"
+)
+
+const (
+	EffectiveAffinity = 1 << iota
+)
+
+type Info struct {
+	Source string
+	IRQ    int
+	CPUs   cpuset.CPUSet
+}
+
+func ReadInfo(procfsRoot string, flags uint) ([]Info, error) {
+	irqRoot := filepath.Join(procfsRoot, "irq")
```

Comment:
Also here, annotate somehow the file we are reading and the format if exits or even link to the kernel code, or even a document you used to understand the contents

Reply:
added reference to the kernel doc